html/includes/authentication/functions.php

```diff
@@ -142,9 +142,12 @@ function set_remember_me()
         dbInsert($db_entry, 'session');
     }
 
-    setcookie('sess_id', $sess_id, $expiration, '/', null, false, true);
-    setcookie('token', $token_id, $expiration, '/', null, false, true);
-    setcookie('auth', $auth, $expiration, '/', null, false, true);
+    // Params: name, value, expire, path, domain, secure, httponly
+    // We set the secure param to $_SERVER["HTTPS"], which will return true
+    // and make it a secure cookie if HTTPS is being used
+    setcookie('sess_id', $sess_id, $expiration, '/', null, $_SERVER["HTTPS"], true);
```

Comment:
I'm not really sure we should just be basing it on if the user is connecting over https or not as surely that's what we have now.

Would it not be better to have the user set in config.php if the connection should be secure and use that variable here?